specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,484 @@
+| `call`                   | Used within a function block to invoke any one of the declared QIS functions and the output recording functions. | May optionally be preceded by a [`tail` marker](https://llvm.org/docs/LangRef.html#call-instruction). |
+| `ret`                    | Used to return the exit code of the program.                                                                     | Must occur (only) as the final instruction of the second (and last) block in the entry point.         |
+| `inttoptr`               | Used to cast an `i64` integer value to either a `%Qubit*` or a `%Result*`.                                       | May be used as part of a function call only.                                                          |
+| `getelementptr inbounds` | Used to create an `i8*` to pass a constant string for the purpose of labeling an output value.                   | May be used as part of call to an output recording function only.                                     |
+
+## Data Types and Values
+
+Within the Base Profile, defining local variables is not supported; instructions
+cannot be nested and constant expressions are fully evaluated. This implies the
+following:
+
+- Call arguments must be constant values, and `inttoptr` casts as well as
+  `getelementptr` instructions must be inlined into a call instruction.
+- It is not possible to express classical computations as part of a Base Profile
+  compliant program.
+
+Constants of any type are permitted as part of a function call. What data types
+occur in the program hence depends entirely on the used QIS functions, as well
+as the runtime functions used for output recording. Constant values of type
+`i64` in particular may be used as part of calls to output recording functions;
+see the section on [output recording](#output-recording) for more details.
+
+For a program to be considered valid, it must make use of at least one qubit and
+perform at least one measurement. The `%Qubit` and `%Result` data types hence
+must be supported by all backends. Qubits and results can occur only as
+arguments in function calls and are represented as a pointer of type `%Qubit*`
+and `%Result*` respectively, where the pointer itself identifies the qubit or
+result value rather than a memory location where the value is stored: a 64-bit
+integer constant is cast to the appropriate pointer type. A more detailed
+elaboration on the purpose of this representation is given in the next
+subsection. The integer constant that is cast must be in the interval `[0,
+nrQubits)` for `%Qubit*` and `[0, nrResults)` for `%Result*`, where `nrQubits`
+and `nrResults` are the required number of qubits and results defined by the
+corresponding [entry point attributes](#attributes). Since backends may look at
```

Comment:
In which case, why not say:
in the interval `[0, required_qubits)` for `%Qubit*` and `[0, required_results)` for `%Result*`

Reply:
I find that confusing; required_qubits is the attribute name, whereas nrQubits represents the attribute value.